dev-tools/generate_notice.py

```diff
@@ -127,8 +129,14 @@ def has_license(folder):
     for subdir in os.listdir(folder):
         if not os.path.isdir(os.path.join(folder, subdir)):
             return False, folder
-        if len(get_licenses(os.path.join(folder, subdir))) == 0:
-            return False, os.path.join(folder, subdir)
+        if len(get_licenses(os.path.join(folder, subdir))) > 0:
+            return True, ""
```

Comment:
this return would avoid checking the rest of subdirs, isn't it?

Reply:
yes, it will check for a licence file first in the parent folder, if found, the function will return True, else, will loop through the subfolders/check/return else sub-subfolders (my addition).

Reply:
I don't fully understand this one. It seems that a single subdir having a license would be enough for this to be valid, what if the rest of subdirs don't have a license?

Reply:
tested the code, it will loop over all subfolders , ex v2, v3, v4 and if they do not have a license it will return false.